.github/workflows/sca.yml

```diff
@@ -70,6 +70,9 @@ jobs:
       - name: Check - phpstan
```

Comment:
do we know how php 7.4 related issue was introduced on master without causing PR to fail first?
imho that incompatibility should be detected already on CI level without any SCA, due to execution of tests on 7.4. Or that means given code path was not covered by tests?

Reply:
@keradus yes, we know, you've introduced it in #7616 when you merged PR without tests and code review 😅.

Reply:
that doesn't actually awnser my question

here, I wonder why the CI was not failing: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7616/commits .

Does it mean `Preg` is not used on 7.4? (which I would be shocked)
Even if specific class that was having wrong typehint is not explicitly tested under 7.4, it is used all around the project, so having full test suite or `php-cs-fixer check` executed, it should show error on wrong typehint. Why it did not (CI for PR was green), do you know?

Reply:
@keradus because the type error only occurs when there's an error during `Preg` usage (that's why I discovered it looking through a failing test case added by @kubawerlos). It would be spotted earlier if there was a test that would involve a return type 😉.